circuit/src/integer/mod.rs

```diff
@@ -250,100 +91,611 @@ where
 		for i in 0..NUM_LIMBS {
 			assigned_result[i] = Some(region.assign_advice(
 				|| format!("result_{}", i),
-				config.x_limbs[i],
+				common.advice[i + NUM_LIMBS],
 				row + 1,
 				|| Value::known(reduction_witness.result.limbs[i]),
 			)?)
 		}
-		let assigned_result = assigned_result.map(|x| x.unwrap());
+		//let assigned_result = assigned_result.map(|x| x.unwrap();
+		let assigned_result = AssignedInteger::new(
+			&reduction_witness.result,
+			&assigned_result.map(|x| x.unwrap()),
+		);
 		Ok(assigned_result)
 	}
+}
+
+/// Chip structure for the IntegerReduce.
+pub struct IntegerReduceChip<
+	W: FieldExt,
+	N: FieldExt,
+	const NUM_LIMBS: usize,
+	const NUM_BITS: usize,
+	P,
+> where
+	P: RnsParams<W, N, NUM_LIMBS, NUM_BITS>,
+{
+	// Assigned integer
+	assigned_integer: AssignedInteger<W, N, NUM_LIMBS, NUM_BITS, P>,
+	/// Constructs phantom datas for the variables.
+	_native: PhantomData<N>,
+	_wrong: PhantomData<W>,
+	_rns: PhantomData<P>,
+}
+
+impl<W: FieldExt, N: FieldExt, const NUM_LIMBS: usize, const NUM_BITS: usize, P>
+	IntegerReduceChip<W, N, NUM_LIMBS, NUM_BITS, P>
+where
+	P: RnsParams<W, N, NUM_LIMBS, NUM_BITS>,
+{
+	/// Creates a new reduce chip
+	pub fn new(assigned_integer: &AssignedInteger<W, N, NUM_LIMBS, NUM_BITS, P>) -> Self {
```

Comment:
Its is a convention to pass values in the constructor instead of references.
Apply in all similar cases